stl/inc/chrono

```diff
@@ -1603,10 +1603,8 @@ namespace chrono {
         constexpr explicit hh_mm_ss(_Duration _Dur)
             : _Is_neg{_Dur < _Duration::zero()}, //
               _Hours{_Duration_cast_underflow_to_zero<_CHRONO hours>(_CHRONO abs(_Dur))},
-              _Mins{_Duration_cast_underflow_to_zero<_CHRONO minutes>(
-                  _Remove_duration_part<_CHRONO hours>(_CHRONO abs(_Dur)))},
-              _Secs{_Duration_cast_underflow_to_zero<_CHRONO seconds>(
-                  _Remove_duration_part<_CHRONO minutes>(_CHRONO abs(_Dur)))} {
+              _Mins{_Duration_cast_underflow_to_zero<_CHRONO minutes>(_CHRONO abs(_Dur)) - _Hours},
+              _Secs{_Duration_cast_underflow_to_zero<_CHRONO seconds>(_CHRONO abs(_Dur)) - _Hours - _Mins} {

```

Comment:
Casting first means we're subtracting integers. `_Dur` is represented as a float but gets cast down immediately.
Casting last means we're subtracting floats. `_Dur` is represented as a float so `_Hours` gets cast back to float (after being previously cast to integer). This is bound to introduce some inaccuracies which we are seeing in the result if the order of operations is changed.
I've written the following code to demonstrate:
```cpp
auto value = duration<float, days::period>{1.55f};
auto hrs = duration_cast<hours>(value);
cout << value.count() << " days" << endl;
cout << endl;
cout << "Casting down first:" << endl;
cout << hrs << " in hours" << endl;
cout << duration_cast<minutes>(value) << " (total) - " << duration_cast<minutes>(hrs) << " (hrs) = ";
cout << duration_cast<minutes>(value) - hrs << endl;
cout << endl;
cout << "Casting down last:" << endl;
cout << value << " (total) - " << duration_cast<decltype(value)>(hrs) << " (hrs) = ";
```
Output:
```
1.55 days

Casting down first:
37h in hours
2232min (total) - 2220min (hrs) = 12min

Casting down last:
1.55d (total) - 1.54167d (hrs) = 0.199997h
11min cast
```

A potential downside would be the days cast to seconds overflowing, however:
If a float isn't used as the input value then it overflows regardless (which is likely the case in 99% of times).
Assuming we're dealing with 64-bit signed ints as the representation for the seconds this would require ~292,271,023,045 years as the input in which case it should be save to say that `hh_mm_ss` provides no valuable information either way which is why I think taking the improved accuracy here is preferable.

Reply:
We unfortunately have zero choice when mandated to use a certain order of operations by the Standard, even if that loses accuracy.

I believe we do need to avoid compile-time overflows (which was apparently a problem when I quickly tried this change) but that's it.

Reply:
Oh, I had assumed the previous implementation was standard compliant concluding this change would be fine too. But this is very understandable, especially considering that other STL implementations seem to exhibit the same loss of accuracy. Too bad the standard dictates this seemingly inferior solution though!
I've pushed a commit that should do the trick. 😄

Reply:
Results after some additional tinkering:
The underflows are manageable by moving `_Duration_cast_underflow_to_zero` implementing it via SFINAE and using it in `operator-` and `+`. (However, this is probably undesirable behavior for general subtraction, so if we go with this we will probably need `_Subtract_with_underflow_to_zero` or something along those lines.)
The overflows however, I am not sure can be solved. I mean, what is the result even supposed to be when trying to cast `hours` to `duration<int, std::ratio<9223372036854775806, 9223372036854775807>`? It can't be expressed as an integer. GCC is also unable to cope with this (although for that matter it also can't cope with the underflows that this implementation can cope with (is handling the underflows even standard compliant? 😄 )).
I guess theoretically the issue lies in `_Fill_tm` and the casting to `hh_mm_ss` isn't actually required for the formatting in this specific case, which means we can either put some kind of bandaid on it or wait until #3575 is resolved which will likely involve a fix for this.
So the options are:
1. Revert to the previous, more accurate calculation which doesn't cause this issue.
2. Implement bandaid fix upon bandaid fix to try and make it stable enough to pass all tests.
3. Wait until #3575 likely takes care of the overflowing call which is not actually needed; after dealing with the underflow.
What are your thoughts?

Reply:
I figured out how to get this to work - we already had the necessary machinery to perform the `- _Hours` and `- _Hours - _Mins` while avoiding underflow. If `_Hours` and `_Mins` couldn't possibly be greater than zero, then our metaprogramming just skips it; e.g. 2<sup>63</sup> - 1 attoseconds isn't even 1 minute.

Reply:
And now I see that I've simply returned the constructor to its original state, just dropping an unnecessary `//` for clang-format, and adding a `_Remove_duration_part<_CHRONO hours>` to follow the Standard more closely (I suspect that skipping it is unobservable, but I don't think it'll hurt).

I've updated the PR description accordingly.